watch.go

```diff
@@ -21,53 +20,116 @@ type Event struct {
 	BatchId uint64
 }
 
-// Watcher temporarily stores event information,
+type WatchChan <-chan *Event
+
+type Watcher interface {
+	Watch() WatchChan
+	Close() error
+}
+
+// watcher temporarily stores event information,
 // as it is generated until it is synchronized to DB's watch.
 //
 // If the event is overflow, It will remove the oldest data,
 // even if event hasn't been read yet.
-type Watcher struct {
-	queue eventQueue
-	mu    sync.RWMutex
+type watcher struct {
```

Comment:
The same name with Water interface, may be set another name for this or interface

Reply:
interface starts with uppercase, the implement struct starts with lowercase, which is conventional.
https://github.com/etcd-io/etcd/blob/70eb293754ae333a1b2e0d761043bb183ad01726/server/storage/mvcc/watcher.go#L37C8-L37C8
https://github.com/etcd-io/etcd/blob/70eb293754ae333a1b2e0d761043bb183ad01726/server/storage/mvcc/watcher.go#L99